cache/mock.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+package cache
```

Comment:
I think it makes sense to start a sub package (perhaps in another PR). We'll probably want somewhere to organize the implementation(s), this will also help ensure we decouple the cache from the provider

Reply:
do you mean a sub-package inside `cache` or inside `provider`?

Reply:
My main concern is ensure we decouple them so the cache can be very dynamic, inside provider works